tests/features/test_covered_distance.py

```diff
@@ -0,0 +1,194 @@
+import unittest
+
+import numpy as np
+import pandas as pd
+
+from databallpy.features.covered_distance import (
+    _parse_intervals,
+    _check_frames,
+    get_covered_distance
+)
+
+
+class TestCoveredDistance(unittest.TestCase):
+    def setUp(self):
+
+        # set input and expected data
+        self.framerate = 1
+        self.vel_intervals = (-1, 13, 30, 17)
+        self.acc_intervals = (37, 15, 0, 0.75)
+
+        self.input = pd.DataFrame(
+            {
+                "home_1_x": [10, 20, -30, 40, np.nan, 60],
+                "home_1_y": [5, 12, -20, 30, np.nan, 60],
+                "home_1_vx": [10, -20, 10, np.nan, 10, np.nan],
+                "home_1_vy": [7, -12.5, 9, np.nan, 15, np.nan],                
+                "home_1_velocity": [
+                    np.sqrt(149), 
+                    np.sqrt(556.25), 
+                    np.sqrt(181), 
+                    np.nan, 
+                    np.sqrt(325), 
+                    np.nan
+                ],
+                "home_1_ax": [-30, 0, np.nan, 0, np.nan, np.nan],
+                "home_1_ay": [-19.5, 1, np.nan, 3, np.nan, np.nan],
+                "home_1_acceleration": [
+                    np.sqrt(1280.25), 
+                    1, 
+                    np.nan, 
+                    3, 
+                    np.nan, 
+                    np.nan
+                ],
+                "away_2_x": [2, 3, 1, 1, 3, 2],
+                "away_2_y": [1, 2, 3, 4, 5, 6],
+                "away_2_vx": [2, 3, 1, 1, 3, 2],
+                "away_2_vy": [1, 1, 1, 1, 1, 1],
+                "away_2_velocity": [
+                    np.sqrt(5), 
+                    np.sqrt(10), 
+                    np.sqrt(2), 
+                    np.sqrt(2), 
+                    np.sqrt(10), 
+                    np.sqrt(5)
+                ],
+                "away_2_ax": [1, -0.5, -1, 1, 0.5, -1],
+                "away_2_ay": [0, 0, 0, 0, 0, 0],
+                "away_2_acceleration": [
+                    1, 
+                    0.5, 
+                    1, 
+                    1, 
+                    0.5, 
+                    1]
+            }
+        )
+
+        self.expected_total_distance = {
+            'home_1': {
+                'total_distance': (np.sqrt(149) + np.sqrt(556.25) + np.sqrt(181) + np.sqrt(325))
+            },
+            'away_2': {
+                'total_distance': (np.sqrt(5) + np.sqrt(10) + np.sqrt(2) + np.sqrt(2) + np.sqrt(10) + np.sqrt(5))
+                }
+        }
+
+        self.expected_output_total_and_vel_distance = {
+            'home_1': {
+                'total_distance': (np.sqrt(149) + np.sqrt(556.25) + np.sqrt(181) + np.sqrt(325)),
```

Comment:
change this to

`'total_distance': self.expected_total_distance["home_1"]`

This will make it easier to make changes if necessary later on without having to change all the differen places where the total distance is calculated. Also, it requires less computation power.